File: python/086_kindergarten-garden/kindergarten_garden.py
class Garden:
    __default_roster = ['Alice', 'Bob', 'Charlie', 'David', 'Eve', 'Fred', 'Ginny', 'Harriet', 'Ileana', 'Joseph', 'Kincaid', 'Larry']
    __encoding_plants = {'G':'Grass', 'C':'Clover', 'R':'Radishes', 'V':'Violets'}
    
    def __init__(self, diagram, students = __default_roster):
        self.__garden = diagram.splitlines()
        self.__roster = sorted(students)

    def plants(self, student):
        index = self.__roster.index(student) * 2
        cup_1 = self.__encoding_plants[self.__garden[0][index]]
        cup_2 = self.__encoding_plants[self.__garden[0][index + 1]]
        cup_3 = self.__encoding_plants[self.__garden[1][index]]
        cup_4 = self.__encoding_plants[self.__garden[1][index + 1]]
        return [cup_1, cup_2, cup_3, cup_4]

File: javascript/ex_118_kindergarten-garden/kindergarten-garden.js
const DEFAULT_STUDENTS = [
  'Alice',
  'Bob',
  'Charlie',
  'David',
  'Eve',
  'Fred',
  'Ginny',
  'Harriet',
  'Ileana',
  'Joseph',
  'Kincaid',
  'Larry',
];

const PLANT_CODES = {
  G: 'grass',
  V: 'violets',
  R: 'radishes',
  C: 'clover',
};

export class Garden {
  constructor(diagram, students = DEFAULT_STUDENTS) {
    this._students = students.sort();
    this._diagram = diagram.split('\n');
  }

  plants(student) {
    let index = this._students.indexOf(student) * 2;
    return [
      PLANT_CODES[this._diagram[0][index]],
      PLANT_CODES[this._diagram[0][index + 1]],
      PLANT_CODES[this._diagram[1][index]],
      PLANT_CODES[this._diagram[1][index + 1]]
    ];
  }
}
